### ISSUE TITLE
Add an option to make parsing erros fatal

### ISSUE THREAD

**Issue by sbraz** [MediaArea/MediaInfoLib#983]
Hi,
Currently, non-existent files cause the library to return an empty info string. I have not read the whole code but looking at examples, I can't find a way to make errors fatal.
This forces me to do stuff like this (which doesn't work on Windows):
https://github.com/sbraz/pymediainfo/blob/69945a82cd9f53a496d33276185cd330c1870d9f/pymediainfo/__init__.py#L199-L204

Would it be possible to add an option to make errors such as non-existent files fatal?

**Comment by JeromeMartinez** [comment]
I agree there is a lot to do on error messages, but here don't get the issue: you use the library, so MediaInfo::Open() returns 0 if there is a problem e.g. the files does not exist.
What do you imagine as a different way to say that there is a problem?

**Comment by sbraz** [comment]
I'm sorry, I completely forgot that… I'm not used to C programming any more 😛
So I guess I should rename this issue to something like "Add error messages when parsing fails"?

**Comment by JeromeMartinez** [comment]
Right. Would be lower priority as there is a workaround for the main issue.

**Comment by sbraz** [comment]
@JeromeMartinez If I understand correctly, the Open function returns the number of general tracks and MediaInfo will *always* find a General track if parsing worked, right?
https://github.com/MediaArea/MediaInfoLib/blob/93f1b2b2cd5788a74ca6bc7f16c40458b1709b1a/Source/MediaInfo/MediaInfo_Internal.cpp#L819

**Comment by JeromeMartinez** [comment]
right ("Parsing worked" = there is a general track also if format is unknown)
